scout/adapter/mongo/omics_variant.py

```diff
@@ -33,20 +33,80 @@ def delete_omics_variants(self, case_id: str, file_type: str):
 
-        samples = []
+        match = {}
+        for ind in case_obj.get("individuals"):
+            if omics_model["sample_id"] == ind.get("rna_sample_id"):
+                match["sample_id"] = ind["rna_sample_id"]
+                match["display_name"] = ind["display_name"]
+                return match
+        return match
+
+    def get_matching_sample_id(self, case_obj: dict, omics_model: dict) -> dict:
+        """
+        Select individual that matches omics model on sample id (as when we are loading a pure RNA case eg).
+        """
 
+        match = {}
         for ind in case_obj.get("individuals"):
             if omics_model["sample_id"] == ind.get("individual_id"):
-                sample = {
-                    "sample_id": ind["individual_id"],
-                    "display_name": ind["display_name"],
-                    "genotype_call": "./1",
-                }
-                samples.append(sample)
+                match["sample_id"] = ind["individual_id"]
+                match["display_name"] = ind["display_name"]
+                return match
+        return match
+
+    def _get_affected_individual(self, case_obj: dict) -> dict:
+        """
+        Fall back to assigning the variants to an individual with affected status to have them display
+        on variantS queries.
+        """
+        match = {}
+        for ind in case_obj.get("individuals"):
+            if ind.get("phenotype") in [2, "affected"]:
+                match["sample_id"] = ind["individual_id"]
+                match["display_name"] = ind["display_name"]
+                return match
+        return match
+
+    def _get_first_individual(self, case_obj: dict) -> dict:
+        """
+        Fall back to assigning the variants to any one individual to have them display
+        on variantS queries.
+        """
+        match = {}
+
+        for ind in case_obj.get("individuals"):
+            match["sample_id"] = ind["individual_id"]
+            match["display_name"] = ind["display_name"]
+            return match
+        return match
+
+    def set_samples(self, case_obj: dict, omics_model: dict):
+        """Internal member function to connect individuals for a single OMICS variant.
+        OMICS variants do not have a genotype as such.
+        Select individuals that match on rna_sample_id if these are provided.
+        For a fallback, match on sample id (as when we are loading a pure RNA case eg),
+        or fall back to assigning the variants to an individual with affected status to have them display
+        on variantS queries.
+        ."""
+
+        samples = []
+
+        match = (
+            self.get_matching_rna_sample_id(case_obj, omics_model)
+            or self.get_matching_sample_id(case_obj, omics_model)
+            or self._get_affected_individual(case_obj)
+            or self._get_first_individual(case_obj)
+        )
+
+        sample = {
+            "sample_id": match["sample_id"],
```

Comment:
```suggestion
            "sample_id": match.get("rna_sample_id",match["sample_id]),
```

if you return the individual in the matching functions

Reply:
I agree it's possible to pass individual, but think this in particular will be difficult to understand without reading the details in the functions above as well? Also the individual is a bit larger data structure. Not so that it wouldn't work, but...

Reply:
You are right, it would only return the name for the data, not all the data. And its potato-potato with reading - it might just reinforce the idea that the `rna_sample_id` is taken first and the `individual_id`after.

Reply:
Plus, hey, it does save lines and memory then!